### PR TITLE
Set fixture method

### PR DESCRIPTION
**Pull request by estrada9166**

### CODE REVIEW

File: lib/tester.js [hunk 1]
```diff
@@ -21,14 +21,24 @@ class Tester {
...
+    this.opts = {}
   }
 
   mock (opts, args = {}) {
     if (!opts || (isObject(opts) && !opts.query && !opts.kind)) {
       throw new Error('The Query/Mutation to test is require')
     }
 
-    let { query, variables, fixture, saveFixture = false } = opts
+    const operationOptions = Object.assign({}, this.opts, opts)
```
Comment (sibelius) [inline lib/tester.js]: you can use spread here

{...this.opts, ...opts}

Comment (estrada9166) [reply lib/tester.js]: Thanks!! 🎉

**Comment by estrada9166** [comment]
`const tester = new EasyGraphQLTester(schema)`

@artola Can you help me check if I'm missing something:
+ Validate fixture against the parsed schema and no against the mock
+ Set fixtures from `tester.setFixture()` used to set them before the test.
+ Set `autoMock` flag on `tester.setFixture(fixture, { autoMock: false })`, used to prevent autoMock of the fields that are not on the fixture.
+ If the `autoMock` is `false` validate that the fields on the fixture match with the request ones on the query (ignore extra fields that are not used on the query)
+ Clear fixture and set `autoMock` to `true` again using `tester.clearFixture()`

### Missing

- [ ] Update documentation

File: utils/fixture.js [hunk 2]
```diff
@@ -1,99 +1,102 @@
...
+      }
 
-    // If it is a scalar, it should be an empty object instead of the __typename
-    if (isObject(mock[val]) && Object.keys(mock[val]).length === 1 && mock[val].__typename) {
-      mock[val] = {}
+      return mock
     }
+  } else if (Array.isArray(fixture)) {
+    fixture.forEach(val => {
+      if (selectedType.isArray && selectedType.noNullArrayValues && val === null) {
```
Comment (artola) [inline utils/fixture.js]: should not be this validation also moved inside `validateType`?

Comment (estrada9166) [reply utils/fixture.js]: Yes! much better!! Thanks!

**Pull request merged**